src/main/java/com/openshift/internal/restclient/capability/resources/ProjectTemplateListCapability.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.internal.restclient.capability.resources;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.internal.restclient.http.UnauthorizedException;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.capability.resources.IProjectTemplateList;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.template.ITemplate;
+
+public class ProjectTemplateListCapability implements IProjectTemplateList {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(IProjectTemplateList.class);
+	private static final String DEFAULT_NAMESPACE = "openshift"; //this will be configurable in near future
+	
+	private IProject project;
+	private IClient client;
+
+	public ProjectTemplateListCapability(IProject project, IClient client) {
+		this.project = project;
+		this.client = client;
+	}
+
+	@Override
+	public boolean isSupported() {
```

Comment:
@adietish I'm not sure I fully understand what you are asking.  It is assumed that we can talk to whatever version of the server we are connected.  Conceptually, you could ask the client the api model version as I think we expose it.  The additional non-null checks are really just sanity checks as they should never be null.  The only case I can think off is if this class is instantiated outside of using our connection logic.